### PR TITLE
Extract move filtering logic out of ClassicBoard

### PR DESCRIPTION
**Pull request by izakfr**
Closes: #120

### CODE REVIEW

File: server/pkg/models/board/board.go [hunk 1]
```diff
@@ -137,6 +141,7 @@ func JoinMoveMaps(left, right MoveMap) {
 	for key := range left {
 		left[key] = append(left[key], right[key]...)
 	}
+
```
Comment (jbsummoner) [inline server/pkg/models/board/board.go:144]: intentional?

File: server/pkg/models/board/classic/entity.go [hunk 2]
```diff
@@ -202,27 +212,27 @@ func (b *ClassicBoard) updateCastlingFlag(move board.Move, sourcePiece *board.Pi
 	if sourcePiece.GetType() == board.KING {
 		switch sourcePiece.GetColor() {
 		case board.BLACK:
-			b.blackAllowedKingsideCastle = false
-			b.blackAllowedQueensideCastle = false
+			b.Disallow(board.KINGSIDE_CASTLE, board.BLACK)
```
Comment (jbsummoner) [inline server/pkg/models/board/classic/entity.go:215]: nit: b.CastlingState.Disallow()

more sematic

File: server/pkg/models/board/castling.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package board
+
+type CastlingState struct {
+	CastlingStateMap map[MoveType]map[Color]bool
+}
+
+func NewCastlingState() *CastlingState {
```
Comment (jbsummoner) [inline server/pkg/models/board/castling.go:7]: should we comment these symbols?

**Review by jbsummoner — APPROVED** [review]

**Comment by jbsummoner** [comment]
I like it :)

File: server/pkg/models/board/board.go (hunk 1 above)
Comment (izakfr) [reply server/pkg/models/board/board.go:144]: Nope

File: server/pkg/models/board/castling.go (hunk 3 above)
Comment (izakfr) [reply server/pkg/models/board/castling.go:7]: Yes, I'll add comments

File: server/pkg/models/board/classic/entity.go (hunk 2 above)
Comment (izakfr) [reply server/pkg/models/board/classic/entity.go:197]: done

File: server/pkg/models/board/castling.go (hunk 3 above)
Comment (izakfr) [reply server/pkg/models/board/castling.go:7]: Done

**Pull request merged**